### PR TITLE
AEA-1223 Add prompt in "aea upgrade" in case project is on registry with newer version

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

With this PR, at the beginning of `aea upgrade`, in case the project is on the registry with a newer version, the user has the option to download the latest version and replace the current project with the new one.

## Fixes

Fixes AEA-1223 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Comment by solarw** [comment]
what about to add -y/--yes flag for command to run it in unattended mode?
this also will help in writing tests, just no need to mock click.confirm

**Review by solarw — APPROVED** [review]
LGTM

**Comment by marcofavorito** [comment]
> what about to add -y/--yes flag for command to run it in unattended mode?
> this also will help in writing tests, just no need to mock click.confirm

I agree we needed the flag `-y`, so I added it in c4ec510.
However, I kept the mocking in the tests as I wanted to test the message printed to the user.

File: aea/cli/upgrade.py [hunk 1]
```diff
@@ -482,8 +576,20 @@ def _prompt(package_id: PackageId, dependencies_to_upgrade: Set[PackageId]):
...
+    """
+    Try to prompt a question to the user.
+
+    The actual effect of this function will be determined by "yes_by_default".
+
+    In particular:
+    - if "yes_by_default" is True, never prompt and return True.
+    - if "yes_by_default" is False, ask to the user.
+    """
+    return click.confirm(message) if not yes_by_default else True
```
Comment (marcofavorito) [inline aea/cli/upgrade.py:592]: notice, if `--yes` is provided, the message is not printed.

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**